src/server/plugins/engine/components/DatePartsField.ts

```diff
@@ -47,7 +45,7 @@ export class DatePartsField extends FormComponent {
           schema: { min: 1, max: 31, precision: 0 },
           options: {
             required: isRequired,
-            optionalText: !isRequired && hideOptional,
+            optionalText: true,
```

Comment:
I've hard coded `optionalText` here rather than a previous workaround to `.replace()` it in the view model